spec/tapioca/compilers/dsl/state_machines_spec.rb

```diff
@@ -0,0 +1,306 @@
+          def alarm_active?; end
+
+          sig { returns(T::Boolean) }
+          def alarm_off?; end
+
+          sig { returns(Integer) }
+          def alarm_state; end
+
+          sig { params(value: Integer).returns(Integer) }
+          def alarm_state=(value); end
+
+          sig { params(state: T.any(String, Symbol)).returns(T::Boolean) }
+          def alarm_state?(state); end
+
+          sig { params(args: T.untyped).returns(T::Array[T.any(String, Symbol)]) }
+          def alarm_state_events(*args); end
+
+          sig { returns(T.any(String, Symbol)) }
+          def alarm_state_name; end
+
+          sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def alarm_state_paths(*args); end
+
+          sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def alarm_state_transitions(*args); end
+
+          sig { params(event: T.any(String, Symbol), args: T.untyped).returns(T::Boolean) }
+          def fire_alarm_state_event(event, *args); end
+
+          sig { returns(String) }
+          def human_alarm_state_name; end
+        end
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for a classe with state_machine, verify path helper method only ") do
+      content = <<~RUBY
+        class Vehicle
+          attr_accessor :seatbelt_on, :time_used, :auto_shop_busy
+
+          state_machine :state, initial: :parked do
+            before_transition parked: any - :parked, do: :put_on_seatbelt
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        sig { params(args: T.untyped).returns(T::Array[::StateMachines::Transition]) }
+          def state_paths(*args); end
+      RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it(" generate RBI for a class with state_machine, verify helper methods start with_ and without") do
+      content = <<~RUBY
+
+        module CustomAttributeIntegration
+          include StateMachines::Integrations::Base
+          def self.integration_name
+            :custom_attribute
+          end
+
+          def create_with_scope(_name)
+            -> {}
+          end
+          def create_without_scope(_name)
+            -> {}
+          end
+        end
+        StateMachines::Integrations.register(CustomAttributeIntegration)
+        class Vehicle
+          state_machine :state, integration: :custom_attribute
+        end
+      RUBY
+
+      expected = <<~RUBY
+        sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def with_state(*states); end
+
+          sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def with_states(*states); end
+
+          sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def without_state(*states); end
+
+          sig { params(states: T.any(String, Symbol)).returns(T.untyped) }
+          def without_states(*states); end
```

Comment:
Can we fix the indentation here please?

Reply:
the indentation is intentional, without it , the test fails.

Reply:
But the indentation between the first line and the subsequent lines are not the same. I doubt that is intentional.

Reply:
As per Kaan  in code explanation, we had to do that to get the test to pass. This is the result of using mixture of  `<<~` and include().